src/vscode-dts/vscode.proposed.tabReveal.d.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// https://github.com/microsoft/vscode/issues/162446
+// https://github.com/microsoft/vscode/issues/148085
+
+declare module 'vscode' {
+
+	export interface TabGroups {
+
+		/**
+		 * Reveals the tab in its current location.
+		 *
+		 * @param tab The tab to reveal.
+		 * @param preserveFocus When `true` focus will remain in its current position.
+		 * @returns A promise that resolves when the tab has been revealed.
+		 */
+		reveal(tab: Tab, preserveFocus?: boolean): Thenable<void>;
```

Comment:
Do you know if this Thenable returns when the didOpen's for a document are finished behind processed or do we need to somehow wait for those?

Reply:
Updated so that it will await IEditorGroup.openEditor.